coresimd/src/x86/i586/sse.rs

```diff
@@ -764,7 +764,7 @@ pub unsafe fn _mm_setzero_ps() -> f32x4 {
 #[inline(always)]
 #[target_feature = "+sse"]
 #[cfg_attr(test, assert_instr(shufps, mask = 3))]
-pub unsafe fn _mm_shuffle_ps(a: f32x4, b: f32x4, mask: i32) -> f32x4 {
+pub unsafe fn _mm_shuffle_ps(a: f32x4, b: f32x4, mask: u32) -> f32x4 {
```

Comment:
It seems that in general the Intel intrinsics "try" to use `unsigned` integers for bit manipulations because that is what makes sense in C++ due to `unsigned` integers having modulo arithmetic (but `signed` integers having undefined behavior on overflow).

The Intel intrinsics don't do this consistently (probably because some of them are 20 years old and this guideline wasn't clear then) but we should probably follow what the new intrinsics do.